### PR TITLE
Add partition iteration resumption test for --use-partition-root=false

### PR DESCRIPTION
**Pull request by Abhi-yb**
### Describe the changes in this pull request

Jenkins run : https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/test-voyager-live-migration-resumption/381/

Adds a long-running partition iteration resumption test that validates live migration with `--use-partition-root=false` for partitioned tables where the root has no primary key but child partitions do (mirrors the Instacart production schema pattern - DB-21087).

The test exercises:
- RANGE-partitioned table (by timestamp) with 5 monthly partitions, 30+ columns including JSONB and sequences
- Forward migration: snapshot + CDC streaming with process kill/restart cycles
- Fallback migration: export-from-target + import-to-source with resumptions
- Cutover to target and cutover to source with restart
- Row count and row hash validation after each iteration
- 20 iterations in CI pipeline (3 locally) with 400s soak periods

Also fixes `segment_hash_validation.sql` to skip partitioned parent tables (`relkind='p'`) that lack primary keys, which previously caused hash validation to fail.

### Describe if there are any user-facing changes

No user-facing changes. This is a test-only addition.

### How was this pull request tested?

- Ran 3 full iterations locally on EC2 (PG source → YugabyteDB target) with all stages passing:
  - Forward: export + import + CDC + resumptions + cutover to target
  - Fallback: export-from-target + import-to-source + resumptions + cutover to source
  - Row count validations: PASS
  - Row hash validations: PASS
- Pipeline template configured for 20 iterations with matching timings from the existing iteration test PR (#3482)

### Does your PR have changes in callhome/yugabyted payloads? If so, is the payload version incremented?

No.

### Does your PR have changes to on-disk structures that can cause upgrade issues?

No.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  user: ${TARGET_DB_USER}
+  password: ${TARGET_DB_PASSWORD}
+  admin:
+    user: ${TARGET_DB_ADMIN_USER}
+    password: ${TARGET_DB_ADMIN_PASSWORD}
+
+voyager:
+  import_data:
+    flags:
+      use-partition-root: "false"
```
Comment (makalaaneesh) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template:136]: the only thing different about this test (as compared to other pre-existing tests) is that we specify this flag use-partition-root=false. Is that correct?

File: migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+            IF batch % 3 = 0 THEN
+                DELETE FROM public.events WHERE id = row_id;
+            END IF;
+        END LOOP;
+
+        IF batch % 50 = 0 THEN
+            PERFORM pg_sleep(0.1);
+        END IF;
+    END LOOP;
```
Comment (makalaaneesh) [inline migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql:44]: should we add the edge-cases that you added to source-dml also to target-dml?

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    action: grant_source_permissions
+    is_live_migration_fall_back: 1
+
+  - name: export_schema_from_source
+    action: voyager_export_schema
+
+  - name: import_schema_on_target
+    action: voyager_import_schema
+
+  - name: loop_start
```
Comment (makalaaneesh) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template:171]: question: is this how we run multiple iterations?

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 4]
```diff
@@ -41,6 +43,46 @@ def __init__(self, cfg: Dict[str, Any], env: Dict[str, str], test_root: str | No
...
         self.process_lock = threading.Lock()
         self.active_resumers: Dict[str, "Resumer"] = {}
+        self.loop_iteration: int = 0
+        self.export_dir_base: str = os.path.abspath(cfg.get("export_dir") or "")
+        self.archive_changes_policy: str | None = None
+        self.prev_archive_file_count: int = 0
+
+
+def apply_effective_export_dir(ctx: Context) -> None:
+    """Point cfg['export_dir'] at the scenario parent for loop_iteration 0, else latest iteration export-dir."""
```
Comment (makalaaneesh) [inline migtests/scripts/live-migration-resumption/helpers.py]: why do we need this? voyager automatically handles the export-dir redirection, right?

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 5]
```diff
@@ -297,11 +338,60 @@ def kill(proc: subprocess.Popen | None, timeout_sec: int = 10) -> None:
...
         proc.kill()
         wait_process(proc, timeout_sec)
+        print(f"kill: killed {proc.pid}")
     except Exception:
+        print(f"kill: failed to kill {proc.pid}")
         # Best effort; ignore
         pass
 
 
+def _kill_pid_graceful(pid: int, timeout_sec: int) -> None:
```
Comment (makalaaneesh) [inline migtests/scripts/live-migration-resumption/helpers.py]: slightly confused. Seems like all these helpers should have already been there before? how come we just had to add it now?

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 6]
```diff
@@ -344,6 +464,18 @@ def _merge_flags(base: Dict[str, Any], extra: Dict[str, Any] | None) -> Dict[str
     return merged
 
 
+def _merge_flags_when_loop_iteration_gte(cfg: Dict[str, Any], op: str, loop_iteration: int) -> Dict[str, Any]:
```
Comment (makalaaneesh) [inline migtests/scripts/live-migration-resumption/helpers.py]: sorry if i'm asking basic questions because i'm not aware of how the framework is built out for iterations: 

Why do we need to merge flags in the first place? Pls add comments to functions.

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 7]
```diff
@@ -483,13 +618,164 @@ def build_import_to_source_replica_cmd(cfg: Dict[str, Any]) -> list[str]:
     return ["yb-voyager", "import", "data", "to", "source-replica", "--yes"] + to_kv_flags(merged)
 
 
+def build_archive_changes_cmd(ctx: Context, policy: str) -> list[str]:
```
Comment (makalaaneesh) [inline migtests/scripts/live-migration-resumption/helpers.py:551]: again, this should have been there earlier as well, no? how come it's showing up in this diff?

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template (hunk 1 above)
Comment (Abhi-yb) [reply migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template:136]: Yes, plus the schema has no PK on the RANGE root (only on children) — mirrors the Instacart pattern where CDC must route to child partitions.

File: migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql (hunk 2 above)
Comment (Abhi-yb) [reply migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql:44]: Sure will add those accordingly..

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template (hunk 3 above)
Comment (Abhi-yb) [reply migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template:171]: Yes — loop_start/loop_end define the loop body, num_iterations controls repeats. Each iteration does forward migration → fallback → validation → cleanup.

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 4 above)
Comment (Abhi-yb) [reply migtests/scripts/live-migration-resumption/helpers.py]: Voyager creates a new export-dir per iteration under live-data-migration-iterations/. The orchestrator needs to track the latest one so wait_for conditions and cutover status check the right path.

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 5 above)
Comment (Abhi-yb) [reply migtests/scripts/live-migration-resumption/helpers.py]: These are from the iteration-test framework (not yet merged to main/RC). Including here since our test depends on it.

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 6 above)
Comment (Abhi-yb) [reply migtests/scripts/live-migration-resumption/helpers.py]: During iteration 0, export runs with snapshot-and-changes (full snapshot + CDC). For iteration 1+, the snapshot already exists so we only need changes-only. flags_when_loop_iteration_gte lets the scenario define these per-iteration overrides declaratively in YAML instead of hardcoding in Python. Will add inline comments.

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 7 above)
Comment (Abhi-yb) [reply migtests/scripts/live-migration-resumption/helpers.py:551]: iteration framework dependency not yet in main.

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 8]
```diff
@@ -0,0 +1,185 @@
+-- Partition iteration test schema
+-- Tests --use-partition-root=false with CDC routing to child partitions.
+-- Three partition types: RANGE (50 partitions, exceeding Instacart's 44), LIST, HASH.
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: no need to add company name in the comment

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+DROP TABLE IF EXISTS public.sales_region CASCADE;
+
+CREATE TABLE public.sales_region (
+    id       SERIAL,
+    amount   INT,
+    branch   TEXT,
+    region   TEXT NOT NULL,
+    metadata JSONB DEFAULT '{}'::jsonb,
+    PRIMARY KEY (id, region)
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: We should keep all these tests with PK on child tables only to cover all the cases with that scenario

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+    dep_code  INT,
+    salary    NUMERIC(10,2),
+    metadata  JSONB DEFAULT '{}'::jsonb
+) PARTITION BY HASH (emp_id);
+
+CREATE TABLE public.emp_0 PARTITION OF public.emp FOR VALUES WITH (MODULUS 5, REMAINDER 0);
+CREATE TABLE public.emp_1 PARTITION OF public.emp FOR VALUES WITH (MODULUS 5, REMAINDER 1);
+CREATE TABLE public.emp_2 PARTITION OF public.emp FOR VALUES WITH (MODULUS 5, REMAINDER 2);
+CREATE TABLE public.emp_3 PARTITION OF public.emp FOR VALUES WITH (MODULUS 5, REMAINDER 3);
+CREATE TABLE public.emp_4 PARTITION OF public.emp FOR VALUES WITH (MODULUS 5, REMAINDER 4);
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql:140]: No PK here?

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+DROP TABLE IF EXISTS public.orders CASCADE;
+
+CREATE TABLE public.orders (
+    order_id    SERIAL,
+    customer_id BIGINT,
+    order_date  DATE NOT NULL,
+    status      TEXT NOT NULL,
+    amount      NUMERIC(10,2),
+    metadata    JSONB DEFAULT '{}'::jsonb,
+    PRIMARY KEY (order_id, order_date, status)
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: same here as mentioned above add PK on childs

File: yb-voyager/src/dbzm/dbzm.go [hunk 12]
```diff
@@ -37,7 +37,7 @@ import (
 var DEBEZIUM_DIST_DIR, DEBEZIUM_CONF_FILEPATH string
 
 // These versions need to be changed at the time of a release
-const DEBEZIUM_VERSION = "2.2.0-1.3.0"
+const DEBEZIUM_VERSION = "0rc1.2.5.2-2026.5.1"
```
Comment (priyanshi-yb) [inline yb-voyager/src/dbzm/dbzm.go]: please revert these two code changes

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 4 above)
Comment (makalaaneesh) [reply migtests/scripts/live-migration-resumption/helpers.py]: 1. cutover status works on the main export-dir. You don't need to use the per-iteration export-dir. 
2. for wait_for conditions, what are we doing here exactly? we should explore if we can use the main export-dir, not the per-iteration export-dir. cc: @priyanshi-yb

File: migtests/tests/resumption/live-migration/partition-iteration-test/iteration_reset.sql [hunk 13]
```diff
@@ -0,0 +1,38 @@
+-- Clean slate between iterations.
```
Comment (makalaaneesh) [inline migtests/tests/resumption/live-migration/partition-iteration-test/iteration_reset.sql]: Why do we reset between iterations? 
Users won't do this. So, our test cases shouldn't either. no?

File: migtests/scripts/event-generator/event-generator.yaml [hunk 14]
```diff
@@ -4,7 +4,7 @@ connection:
   # Port PostgreSQL is listening on
   port: 5432
   # Target database name
-  database: sakila
```
Comment (priyanshi-yb) [inline migtests/scripts/event-generator/event-generator.yaml:7]: I believe these are just for some local testing? Can you please remove them if not required

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 15]
```diff
@@ -0,0 +1,382 @@
...
+    connection:
+      host: 127.0.0.1
+      port: 5432
+      database: test_db
+      user: postgres
+      password: postgres
+
+    generator:
+      schema_name: public
+      manual_table_list: [events, sales_region, emp, orders]
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: I don't think you need to specify the table list. If it's not specified, it fetches the list of tables directly from the database, which should be fine.

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 16]
```diff
@@ -0,0 +1,382 @@
...
+    connection:
+      host: ${TARGET_DB_HOST}
+      port: ${TARGET_DB_PORT}
+      database: test_db
+      user: ${TARGET_DB_USER}
+      password: ${TARGET_DB_PASSWORD}
+
+    generator:
+      schema_name: public
+      manual_table_list: [events, sales_region, emp, orders]
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: same

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 17]
```diff
@@ -0,0 +1,382 @@
...
+  import_data:
+    flags:
+      use-partition-root: "false"
+  import_to_source:
+    flags:
+      use-partition-root: "false"
+  export_data:
+    flags_when_loop_iteration_gte:
+      "1":
+        export-type: changes-only
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: remove this as discussed for previous PR

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 18]
```diff
@@ -0,0 +1,382 @@
...
+    action: row_hash_validations
+
+  - name: cleanup_source
+    action: run_sql
+    sql_path: ./iteration_reset.sql
+
+  - name: cleanup_target
+    action: run_sql
+    target: target
+    sql_path: ./iteration_reset.sql
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: similarly remove these also

File: migtests/tests/resumption/live-migration/partition-iteration-test/source_dml.sql [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+        (2000, 'Harbor Rd', 'Sydney', '{"type": "wholesale"}'::jsonb),
+        (3000, 'Beacon St', 'Boston', '{"type": "retail"}'::jsonb),
+        (4000, 'Shibuya', 'Tokyo', '{"type": "flagship"}'::jsonb),
+        (5000, 'Unter den Linden', 'Berlin', '{"type": "outlet"}'::jsonb),
+        (6000, 'Unknown Blvd', 'Mumbai', '{"type": "new_market"}'::jsonb);  -- lands in DEFAULT
+
+    -- UPDATE on LIST partition
+    UPDATE public.sales_region SET amount = amount + 500, metadata = metadata || '{"updated": true}'::jsonb
+    WHERE region = 'London';
+
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/source_dml.sql:107]: add a change partition case here similar to this  
```
 -- TC17: Cross-partition UPDATE — INSERT into Jun 2024, then UPDATE created_at to Jul 2024
    -- PG internally does DELETE from events_202406 + INSERT into events_202407
    -- CDC must replicate this as a DELETE+INSERT pair on the correct child partitions
```

File: migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    -- ===================== LIST TABLE: edge cases =====================
+
+    INSERT INTO public.sales_region (id, amount, branch, region, metadata) VALUES
+        (90001, 7000, 'Target Branch', 'Sydney', '{"target": true}'::jsonb),
+        (90002, 8000, 'Target Branch 2', 'Tokyo', '{"target": true}'::jsonb),
+        (90003, 9000, 'Target Default', 'Paris', '{"target": true}'::jsonb);  -- DEFAULT
+
+    UPDATE public.sales_region SET amount = 9999, metadata = metadata || '{"tgt_updated": true}'::jsonb
+    WHERE id = 90001 AND region = 'Sydney';
+
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/target_dml.sql:89]: same as above

File: yb-voyager/src/utils/version.go [hunk 21]
```diff
@@ -17,7 +17,7 @@ package utils
 
 const (
 	// This constant must be updated on every release.
-	YB_VOYAGER_VERSION = "main"
+	YB_VOYAGER_VERSION = "0rc1.2026.5.1"
```
Comment (priyanshi-yb) [inline yb-voyager/src/utils/version.go]: please revert this change

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 22]
```diff
@@ -0,0 +1,367 @@
...
+    graceful_timeout_sec: 60
+
+  - name: sleep
+    action: sleep
+    seconds: 10
+
+  - name: clear_cutover_table_on_target
+    action: run_sql
+    target: target
+    sql_path: ../common-sql/delete_marker.sql
```
Comment (priyanshi-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: I don't think we need this now right?

File: migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template [hunk 23]
```diff
@@ -0,0 +1,367 @@
...
+
+  - name: import_schema_on_target
+    action: voyager_import_schema
+
+  - name: loop_start
+    action: loop_start
+
+  - name: stop_prev_iteration_exporter
+    action: voyager_stop_command
+    command: export_data
```
Comment (makalaaneesh) [inline migtests/tests/resumption/live-migration/partition-iteration-test/scenario.yaml.template]: export_data_from_target, import_data_to_source

File: migtests/scripts/event-generator/generator.py [hunk 24]
```diff
@@ -151,41 +154,45 @@
...
 
                 success = execute_with_retry(run_once, rebuild, conn.rollback, max_retries=INSERT_MAX_RETRIES)
                 if success:
                     # Placeholder: hook for future stats reporting (e.g., counting successful INSERT batches)
                     pass
             
             elif operation == "UPDATE":
+                primary_key = table_schemas[table_name]["primary_key"]
+                if not primary_key:
+                    print(f"Skipping UPDATE on '{table_name}': no primary key found")
```
Comment (makalaaneesh) [inline migtests/scripts/event-generator/generator.py:175]: just checking: this means we will skip updates on the parent, but we will still issue UPDATEs on the child tables, correct?

File: migtests/scripts/event-generator/generator.py [hunk 25]
```diff
@@ -199,12 +206,17 @@ def rebuild():
...
-                        break  # Break out of the loop if the update is successful
+                        break
                     except Exception as e:
+                        print(f"UPDATE failed on '{table_name}': {e}")
                         conn.rollback()
 
             elif operation == "DELETE":
                 primary_key = table_schemas[table_name]["primary_key"]
+                if not primary_key:
+                    print(f"Skipping DELETE on '{table_name}': no primary key found")
```
Comment (makalaaneesh) [inline migtests/scripts/event-generator/generator.py:217]: same question: we will have DELETEs on the child tables, right?

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 26]
```diff
@@ -364,9 +363,9 @@ def kill(proc: subprocess.Popen | None, timeout_sec: int = 10) -> None:
             return
         proc.kill()
         wait_process(proc, timeout_sec)
-        print(f"kill: killed {proc.pid}")
+        log(f"kill: killed {proc.pid}")
     except Exception:
-        print(f"kill: failed to kill {proc.pid}")
+        log(f"kill: failed to kill {proc.pid}")
```
Comment (shubham-yb) [inline migtests/scripts/live-migration-resumption/helpers.py]: Where does this log? Let's revert this change if not required.

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 27]
```diff
@@ -0,0 +1,272 @@
...
+-- TC18: Empty partition — validates Voyager handles partitions with zero rows
+CREATE TABLE public.events_202606 PARTITION OF public.events
+    FOR VALUES FROM ('2026-06-01 00:00:00') TO ('2026-07-01 00:00:00');
+ALTER TABLE public.events_202606 ADD PRIMARY KEY (id);
+
+-- TC38: DEFAULT partition — catches rows outside all defined ranges
+CREATE TABLE public.events_default PARTITION OF public.events DEFAULT;
+ALTER TABLE public.events_default ADD PRIMARY KEY (id);
+
+-- REPLICA IDENTITY FULL on root + all children for CDC without PK on root
```
Comment (shubham-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: Does the test framework not handle setting replica identity for all tables?

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql [hunk 28]
```diff
@@ -0,0 +1,272 @@
...
+CREATE TABLE public.sales_default PARTITION OF public.sales_region DEFAULT;
+
+ALTER TABLE public.sales_london  ADD PRIMARY KEY (id, region);
+ALTER TABLE public.sales_sydney  ADD PRIMARY KEY (id, region);
+ALTER TABLE public.sales_boston   ADD PRIMARY KEY (id, region);
+ALTER TABLE public.sales_tokyo   ADD PRIMARY KEY (id, region);
+ALTER TABLE public.sales_berlin  ADD PRIMARY KEY (id, region);
+ALTER TABLE public.sales_default ADD PRIMARY KEY (id, region);
+
+ALTER TABLE public.sales_region REPLICA IDENTITY FULL;
```
Comment (shubham-yb) [inline migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: Same question as above.

File: migtests/scripts/event-generator/generator.py (hunk 24 above)
Comment (Abhi-yb) [reply migtests/scripts/event-generator/generator.py:175]: Generator resolves PK from children for parent tables (no PK on root). It also discovers child partitions directly (they have PKs). So DML happens on both parents and children — both valid paths for CDC capture

File: migtests/scripts/event-generator/generator.py (hunk 25 above)
Comment (Abhi-yb) [reply migtests/scripts/event-generator/generator.py:217]: Same as UPDATE — generator resolves PK from children for parent tables, so DELETE goes through the parent. PG routes it to the right child. Child partitions also get direct DELETEs since they have their own PKs.

File: migtests/scripts/live-migration-resumption/helpers.py (hunk 4 above)
Comment (Abhi-yb) [reply migtests/scripts/live-migration-resumption/helpers.py]: Voyager commands always use the main export-dir. The helper only tracks the per-iteration export-dir internally for queue scans and archive validation checks.

File: migtests/tests/resumption/live-migration/partition-iteration-test/init.sql (hunk 27 above)
Comment (Abhi-yb) [reply migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: No, the framework doesn't handle it

Comment (shubham-yb) [reply migtests/tests/resumption/live-migration/partition-iteration-test/init.sql]: Our grant script already handles it. Do check.

**Review by shubham-yb — APPROVED** [review]
LGTM

File: migtests/scripts/live-migration-resumption/helpers.py [hunk 29]
```diff
@@ -588,25 +587,11 @@ def _return_segment_files(directory: str) -> list[str]:
...
-    with open(filepath, "r") as f:
-        for line in f:
-            line = line.strip()
-            if not line or line == r"\.":
-                continue
-            if first is None:
-                first = json.loads(line)["vsn"]
-            last_line = line
-    last = json.loads(last_line)["vsn"] if last_line is not None else None
-    return first, last
```
Comment (priyanshi-yb) [inline migtests/scripts/live-migration-resumption/helpers.py:604]: why have we removed it? are we not using this?

**Review by priyanshi-yb — APPROVED** [review]
LGTM!
